src/shipit_static_analysis/shipit_static_analysis/workflow.py

```diff
@@ -111,14 +107,51 @@ def run(self, revision, review_request_id, diffset_revision):
 
         # Run static analysis through run-clang-tidy.py
         logger.info('Run clang-tidy...')
-        issues = self.clang.run(self.config['clang_checkers'], modified_files)
+        issues = self.clang.run(settings.clang_checkers, modified_files)
 
         logger.info('Detected {} code issue(s)'.format(len(issues)))
+        if not issues:
+            logger.info('No issues, stopping there.')
+            return
+
+        # Publish on mozreview
+        self.publish_mozreview(review_request_id, diffset_revision, issues)
 
         # Notify by email
-        if issues:
-            logger.info('Send email to admins')
-            self.notify_admins(review_request_id, issues)
+        logger.info('Send email to admins')
+        self.notify_admins(review_request_id, issues)
+
+    def publish_mozreview(self, review_request_id, diff_revision, issues):
+        '''
+        Publish comments on mozreview
+        '''
+
+        # Filter issues to keep publishable checks
+        # and non third party
+        def _is_publishable(issue):
+            return issue.has_publishable_check() \
+                and not issue.is_third_party()
+
+        issues = list(filter(_is_publishable, issues))
+        if not issues:
+            logger.info('No issues to publish on MozReview')
+            return
+
+        # Create batch review
+        review = BatchReview(
+            self.mozreview,
+            review_request_id,
+            diff_revision,
+        )
+
+        # Comment each issue
+        for issue in issues:
+            logger.info('Will publish about {}'.format(issue))
+            review.comment(issue.path, issue.line, 1, issue.body)
```

Comment:
I'm not sure how MozReview handles comments on lines that were not actually modified by a changeset (e.g. when we find code issues that are in a file modified by the patch, but in a part of the file that was actually changed by the patch).

Shouldn't we check that the comments we add all associate to a line that was actually modified by the patch? And/or adjust lines with offsets like in [this example code](https://hg.mozilla.org/hgcustom/version-control-tools/file/0acfc2c6fd8e/pylib/mozreviewbots/pylintbot/__main__.py#l142)? (Note: I don't really know what this code does actually, we should ask @smacleod about it).